frontend/docsettings.lua

```diff
@@ -276,7 +276,7 @@ function DocSettings:flush(data, no_custom_metadata)
                     end
                 end
                 -- custom metadata
-                metadata_file = DocSettings.getCustomMetadataFile(self.data.doc_path)
+                metadata_file = self:getCustomMetadataFile(self.data.doc_path)
```

Comment:
Haven't checked context, but passing a `self.` something arg to a method feels... weird?

Can't the method itself just dig into its insides for the right thing directly?

Reply:
(Especially a semi-private self.data as far as LuaSettings objects are concerned)